template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerLdapConfigProvider.java

```diff
@@ -0,0 +1,192 @@
+                        configVar(RANGER_USERSYNC_LDAP_LDAPBINDPASSWORD, getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_LDAP_LDAPBINDPASSWORD)),
+                        configVar(RANGER_USERSYNC_LDAP_USER_NAMEATTRIBUTE,
+                                getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_LDAP_USER_NAMEATTRIBUTE)),
+                        configVar(RANGER_USERSYNC_LDAP_USER_SEARCHBASE, getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_LDAP_USER_SEARCHBASE)),
+                        configVar(RANGER_USERSYNC_LDAP_USER_OBJECTCLASS, getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_LDAP_USER_OBJECTCLASS)),
+                        config("ranger.usersync.ldap.deltasync", Boolean.FALSE.toString()),
+                        config("ranger.usersync.group.searchenabled", Boolean.TRUE.toString()),
+                        configVar(RANGER_USERSYNC_GROUP_MEMBERATTRIBUTENAME,
+                                getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_GROUP_MEMBERATTRIBUTENAME)),
+                        configVar(RANGER_USERSYNC_GROUP_NAMEATTRIBUTE, getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_GROUP_NAMEATTRIBUTE)),
+                        configVar(RANGER_USERSYNC_GROUP_OBJECTCLASS, getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_GROUP_OBJECTCLASS)),
+                        configVar(RANGER_USERSYNC_GROUP_SEARCHBASE, getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_GROUP_SEARCHBASE)),
+                        configVar(RANGER_USERSYNC_GROUP_SEARCHFILTER, getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_GROUP_SEARCHFILTER)),
+                        configVar(RANGER_USERSYNC_GROUP_BASED_ROLE_ASSIGNMENT_RULES,
+                                getRoleTypeVariableName(hostGroup, roleType, RANGER_USERSYNC_GROUP_BASED_ROLE_ASSIGNMENT_RULES))
+                );
+                // Note: the following settings are not required:
+                // ranger.usersync.ldap.searchBase
+                // ranger.usersync.ldap.user.searchfilter
+                // ranger.usersync.ldap.user.nameattribute
+                // ranger.usersync.ldap.user.groupnameattribute
+            default:
+                return List.of();
+        }
+    }
+
+    @Override
+    protected List<ApiClusterTemplateVariable> getRoleConfigVariable(String roleType, HostgroupView hostGroupView, TemplatePreparationObject source) {
+        String hostGroup = hostGroupView.getName();
+        LdapView ldapView = source.getLdapConfig().get();
+        switch (roleType) {
+            case RangerRoles.RANGER_ADMIN:
+                return List.of(
+                        variable(getRoleTypeVariableName(hostGroup, roleType, RANGER_AUTHENTICATION_METHOD), ldapView.getDirectoryType().toString()),
+                        variable(getRoleTypeVariableName(hostGroup, roleType, RANGER_LDAP_AD_DOMAIN), " "),
```

Comment:
Is the value `" "` intended or a typo?

Reply:
I do not know for sure. 😃 That was taken from the original Ambari counterpart (https://github.com/hortonworks/cloudbreak/blob/master/template-manager-blueprint/src/main/resources/blueprints/configurations/ranger/ldap.handlebars). In fact, I doubt this would work correctly with an AD as is, as one would usually expect `getDomain()` here.

Reply:
Okay. Just checked in the Ranger CM page.
> AD domain. Only used if Authentication method is AD.

Setting this property via CM and then restarting Ranger did not hurt, so the code will be fixed to set this according to `getDomain()`.

Reply:
Fixed.